src/components/ha-code-editor.ts

```diff
@@ -94,7 +94,7 @@ export class HaCodeEditor extends ReactiveElement {
 
   public disconnectedCallback() {
     super.disconnectedCallback();
-    this.removeEventListener("keydown", stopPropagation);
+    this.codemirror?.destroy();
```

Comment:
I don't think it's needed because it's attached to the element itself. If the element is disconnected, listeners automatically go away to right?

Reply:
We usually remove then manually but that's a good point that need to be tested. Not sure it's needed.

Reply:
Okay I just poked around and there is a good mix of removing and not removing throughout the code.  Trend seems to be that newer stuff does and older doesn't. 🤷🏻 

I just added it back here and we can police all of them as a follow up I guess.

Reply:
If an element is removed from the DOM, event listeners are not removed and since there still is a reference to the element it will never get garbage collected, thus it is a memory leak.

Reply:
Sure that's true if the listener is attached outside the host element, but if the listener is on the host  or its children, then it should be removed as well.  I haven't tested this myself though - just going off a [statement in the lit docs](https://lit.dev/docs/components/lifecycle/#use-cases-2):

> No need to remove internal event listeners. You don't need to remove event listeners added on the component's own DOM—including those added declaratively in your template. Unlike external event listeners, these won't prevent the component from being garbage collected.